src/raftstore/store/peer.rs

```diff
@@ -459,13 +462,40 @@ impl Peer {
             ready.hs.take();
         }
 
+        if let Some(ref soft_state) = ready.ss {
+            if soft_state.leader_id == raft::INVALID_ID {
+                if self.leader_missing_time.is_none() {
+                    self.leader_missing_time = Some(Instant::now())
+                }
+            } else if self.is_initialized() {
```

Comment:
> In raft, if the follower doesn't receive the message from leader after ElectionTimeout, it will begin to campaign, and clear the leader.

It's true for initialized peers, which already has all nodes' info of the cluster.